src/Api/AdminConsole/Queries/OrganizationUserUserDetailsQuery.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Data.Organizations.OrganizationUsers;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+
+namespace Api.AdminConsole.Queries;
+
+// Request object. If we need to do the Auth check in the query (see below). 
+// The ClaimsPrincipal would need to be passed in from the endpoint. I see the
+// AuthorizeAsync sort of does the same thing with the ClaimsPrincipal for passing
+// to the service so it might not be a big deal
+public class OrganizationUserUserDetailsQueryRequest
+{
+    // Again if this is necessary. Wanted to name it something so 
+    // it's clear it is not the Bitwarden user info. Rather the ClaimsPrincipal
+    // public ClaimsPrincipal RequestingUserContext { get; set; }
+    public Guid OrganizationId { get; set; }
+    public bool IncludeGroups { get; set; } = false;
+    public bool IncludeCollections { get; set; } = false;
+}
+
+public class OrganizationUserUserDetailsQueryResponse
+{
+    public OrganizationUserUserDetails OrganizationUserUserDetails { get; set; }
+    public bool TwoFactorEnabled { get; set; }
+    public Permissions Permissions { get; set; }
+
+    public OrganizationUserUserDetailsQueryResponse(
+        OrganizationUserUserDetails organizationUserUserDetails,
+        bool twoFactorEnabled
+    )
+    {
+        this.OrganizationUserUserDetails = organizationUserUserDetails;
+        this.TwoFactorEnabled = twoFactorEnabled;
+        // Convert the OrganizationUserUserDetails permissions json string 
+        this.Permissions = CoreHelpers.LoadClassFromJsonData<Permissions>(OrganizationUserUserDetails.Permissions);
+    }
+}
+
+public class OrganizationUserUserDetailsQuery : IOrganizationUserUserDetailsQuery
+{
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IUserService _userService;
+
+    public OrganizationUserUserDetailsQuery(
+        IOrganizationUserRepository organizationUserRepository,
+        IUserService userService
+    )
+    {
+        _organizationUserRepository = organizationUserRepository;
+        _userService = userService;
+    }
+
+    public async Task<IEnumerable<OrganizationUserUserDetailsQueryResponse>> GetOrganizationUserUserDetails(OrganizationUserUserDetailsQueryRequest request)
+    {
+        // Should the query authorize? Or should this be the responsibility of the controller? 
+        // Maybe having the query authorize would ensure safety?
+        // Code from the controller for auth:
+        // var authorized = (await _authorizationService.AuthorizeAsync(
+        //     User, OrganizationUserOperations.ReadAll(orgId))).Succeeded;
+        // if (!authorized)
+        // {
+        //     throw new NotFoundException();
+        // }
```

Comment:
For now at least, I think this should continue to live in the controller; our private and public api use different authorization patterns (private uses a mix of handrolled and resource-based authorization, public uses policy-based authorization), and introducing them at the query level tightly couples the query logic with this particular authorization pattern.

Maybe there's some opportunity to align this in the future, I'm not sure.